packages/flutter/lib/src/rendering/viewport.dart

```diff
@@ -1399,7 +1399,7 @@ class RenderViewport extends RenderViewportBase<SliverPhysicalContainerParentDat
     return constraints.biggest;
   }
 
-  static const int _maxLayoutCycles = 10;
+  static const int _maxLayoutCyclesPerChild = 10;
```

Comment:
Should this be a little lower now that it is per child? Before a viewport with 3 sliver children would allow 10 passes, now it would be 30.

Reply:
I don't think any reasonable sliver would need more than a handful of passes, the problem is that you can have nested slivers in [MultiSliver](https://pub.dev/documentation/sliver_tools/latest/sliver_tools/MultiSliver-class.html) or even `SliverMainAxisGroup` (but I don't think that one properly propagates `scrollOffsetCorrection`?) and we have no good way to count correction count per actual source sliver. For that we would probably need changes to `SliverGeometry`.

So lowering the `mainLayoutCycles` could potentially be breaking change for someone that has nested slivers.

More importantly I do think that the check is meant to detect pathological cases where the layout is stuck in a loop of corrections, and as such it does matter if it is 10 passes or 30? It will throw an exception anyway so it is not something that happens during regular layout pass.

Reply:
SGTM, thanks for explaining your thinking here. :)